modules/messages/client/components/Inbox.component.js

```diff
@@ -0,0 +1,134 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { useTranslation } from 'react-i18next';
+
+import * as api from '../api/messages.api';
+import Avatar from '@/modules/users/client/components/Avatar.component';
+import Activate from '@/modules/users/client/components/Activate';
+import { $broadcast, eventTrack } from '@/modules/core/client/services/angular-compat';
+import TimeAgo from '@/modules/core/client/components/TimeAgo';
+
+const userType = PropTypes.shape({
+  _id: PropTypes.string.isRequired,
+  username: PropTypes.string.isRequired,
+  displayName: PropTypes.string.isRequired,
+});
+
+export default function Inbox({ user }) {
+  if (!user.public) {
+    return (
+      <section className="container-spacer">
+        <Activate/>
+      </section>
+    );
+  }
+  const { t } = useTranslation('messages');
+
+  const [nextParams, setNextParams] = useState(null);
+  const [isFetching, setIsFetching] = useState(false);
+  const [threads, setThreads] = useState([]);
+
+  const hasMore = Boolean(nextParams);
+
+  async function fetchThreads(next = false) {
+    setIsFetching(true);
+    try {
+      if (next) {
+        eventTrack('inbox-pagination', {
+          category: 'messages.inbox',
+          label: 'Inbox page ' + nextParams,
+        });
+      }
+
+      const data = await api.fetchThreads(next ? nextParams : {});
+      setThreads(threads => next ? threads.concat(data.threads) : data.threads);
+      setNextParams(data.nextParams);
+    } finally {
+      setIsFetching(false);
+    }
+  }
+
+  useEffect(() => {
+    // Tell unread-messages directive to sync itself
+    $broadcast('syncUnreadMessagesCount');
+    fetchThreads();
+  }, []);
+
+  return <section className="container-spacer">
+    {!isFetching && threads.length === 0 && (
+      <div className="content-empty">
+        <i className="icon-3x icon-messages-alt"/>
+        <h4 role="alert">{t('No conversations yet.')}</h4>
+      </div>
```

Comment:
We can replace this with a component from https://github.com/Trustroots/trustroots/pull/1162 later on.

Reply:
Yup! I'll wait until then... where we can revisit the `role` thing... I'll keep it as _alert_ for now.